File: Labs/FinalExamReference/recursion/flattenNestedLoop/flatten.m
% Octave - uses cell arrays
function result = flatten(lst)
    if isempty(lst)
        result = {};
    elseif iscell(lst{1})
        result = [flatten(lst{1}), flatten(lst(2:end))];
    else
        result = [lst(1), flatten(lst(2:end))];
    end
end

File: Labs/FinalExamReference/recursion/flattenNestedLoop/flatten.py
# Python
def flatten(lst):
    if not lst: return []
    if isinstance(lst[0], list):
        return flatten(lst[0]) + flatten(lst[1:])
    return [lst[0]] + flatten(lst[1:])

print(flatten([1, [2, [3, 4]], 5]))     # [1, 2, 3, 4, 5]